data_iterator.go

```diff
@@ -165,11 +166,25 @@ func (d *DataIterator) Run(tables []*TableSchema) {
 		loggingIncrement = 1
 	}
 
-	for table, _ := range tablesWithData {
-		tablesQueue <- table
+	type kv struct {
+		Table *TableSchema
+		MaxPaginationKey uint64
+	}
+
+	sortedTableData := make([]kv, 0, len(tablesWithData))
+	for k, v := range tablesWithData {
+		sortedTableData = append(sortedTableData, kv{k, v})
+	}
+
+	sort.Slice(sortedTableData, func(i, j int) bool {
+		return sortedTableData[i].MaxPaginationKey > sortedTableData[j].MaxPaginationKey
```

Comment:
This is fine for now, but I'm wondering if we can get a bit more accurate via `information_schema`. Specifically, we can estimate a table size via something like:

```
SELECT 
    table_name AS `Table`, 
    round(((data_length + index_length) / 1024 / 1024), 2) `Size in MB` 
FROM information_schema.TABLES 
WHERE table_schema = "$DB_NAME"
    AND table_name = "$TABLE_NAME";
```

In our case, I think we only need to know about the `data_length` instead of the `data_length + index_length`. However, this could still be a relatively inaccurate estimate if one uses ghostferry-sharding to select a subset of data.

Ultimately, I think a good idea would be to have an interface in Ghostferry that allows application to implement arbitrary table sorting algorithms, so the code could look like something like this:

```
// bad name, but just an example
// Alternatively, this method could be in the TableFilter struct 
type TableSorterForDataIterator interface { 
  Sort(tables []*TableSchema) []*TableSchema // Maybe a better name here..
}
```

A custom table sorter implementing that interface is passed to Ferry then passed to DataIterator, which is then used here. We can implement a few simple algorithm within Ghostferry (based on both information_schema and naively via max_primary_key) so people can choose from that if they don't want to implement their own.

This would be in line with how we do things with Filters, Thottlers, and Verifier.